docs/usage/type-conversions.md

```diff
@@ -326,6 +329,115 @@ Javascript map can use `true` and `1` as distinct keys but a Python `dict`
 cannot. If the Javascript map contains both `true` and `1` a `ConversionError`
 will be thrown.
 
+## Functions
+
+### Calling Python objects from Javascript
+
+If a Python object is callable, the proxy will be callable too. The arguments
+will be translated from Javascript to Python as appropriate, and the return
+value will be translated from Javascript back to Python. If the return value is
+a `PyProxy`, you must explicitly destroy it or else it will be leaked.
+
+An example:
+
+```pyodide
+let test = pyodide.runPython(`
+    def test(x):
+        return [n*n for n in x]
+    test
+`);
+let result_py = test([1,2,3,4]);
+// result_py is a PyProxy of a list.
+let result_js = result_py.toJs();
+// result_js is the array [1, 4, 9, 16]
+result_py.destroy();
+```
+
+If a function is indended to be used from Javascript, you can use {any}`to_js <pyodide.to_js>` on the return value. This is particularly important for
+callbacks.
+
+```pyodide
+let test = pyodide.runPython(`
+    from pyodide import to_js
+    def test(x):
+        return to_js([n*n for n in x])
+    test
+`);
+let result = test([1,2,3,4]);
+// result is the array [1, 4, 9, 16], nothing needs to be destroyed.
+```
+
+If you need to use a key word argument, use {any}`callKwargs <PyProxy.callKwargs>`. The last argument should be a Javascript object with the
+key value arguments.
+
+```pyodide
+let test = pyodide.runPython(`
+    from pyodide import to_js
+    def test(x, *, offset):
+        return to_js([n*n + offset for n in x])
+    to_js(test)
+`);
+let result = test.callKwargs([1,2,3,4], { offset : 7});
+// result is the array [8, 12, 16, 23]
+```
+
+(call-js-from-py)=
+
+### Calling Javascript functions from Python
+
+What happens when calling a Javascript function from Python is a bit more
+complicated than calling a Python function from Javascript. If there are any
+keyword arguments, they are combined into a Javascript object and used as the
+final argument. Thus, if you call:
+
+```py
+f(a=2, b=3)
+```
+
+then the Javascript function receives one argument which is a Javascript object
+`{a : 2, b : 3}`.
+
+When a Javascript function is called and it returns anything but a promise, if
+the result is a `PyProxy` it is destroyed. Also, any arguments that are
+PyProxies that were created in the process of argument conversion are also
+destroyed. If the `PyProxy` was created in Python using
+{any}`pyodide.create_proxy` it is not destroyed.
+
+When the Javascript function returns a `Promise` (for example, if the function
+is an `async` function), the behavior is more complicated. In this case, the
```

Comment:
```suggestion
is an `async` function), the behavior is more complicated to minimize memory leaks. 
In this case, the
```

Reply:
Well it's sort of the opposite: we destroy stuff to avoid memory leaks. If a `Promise` is returned, there is a high chance the arguments are still needed for whatever work the `Promise` is going to do and so that destroying them will cause trouble. So when you return a Promise the lifetime of the proxies is extended from "for the duration of this function call" to "until this `Promise` resolves".